src/libraries/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs

```diff
@@ -709,6 +709,20 @@ private static WinHttpChunkMode GetChunkedModeForSend(HttpRequestMessage request
             return chunkedMode;
         }
 
+        private static bool ContainsOnlyValidLatin1(string headerString)
+        {
+            foreach (char c in headerString)
+            {
+                // See https://www.rfc-editor.org/rfc/rfc9110.html#section-5.5-5:
+                // "Field values containing CR, LF, or NUL characters are invalid and dangerous"
+                if (c <= 0 || c > 255 || c == '\r' || c == '\n')
```

Comment:
```suggestion
                if (c is <= 0 or > 255 or '\r' or '\n')
```
In the future Roslyn may be able to optimize this with pattern matching.